### DISCUSSION TITLE
Trying to figure out where to start

### DISCUSSION THREAD

**Discussion by Ianlucht** [ppy/osu#16759] [Q&A]
Hello, I am a freshman computer science major at Colorado State University. This is my first time with an open-source project, and I have to say I am a bit overwhelmed, if I could just have a little guidance over where a good place would be to start getting used to participating in the open source eco system that would be great.

**Comment by mk56-spn** [comment]
do you play lazer aside from wanting to code for it? if so id start with something small that you feel is missing, ( probably something quality of life type) by opening a discussion in the idea category and seeing if its something thats required / wanted, since you are a computer major im gonna guess that you are already pretty competent in c# so osu!framework will be probably the hardest part for you ( just guessing so sorry if im wrong) so id spend some time trying to get comfortable with that, 

a good way to do this imo is just trying to remake some function or skim through the tests  while they are running to get a better idea of how it all works, if its something graphical then  hitting ctrl+f1 whilst in lazer will allow you to see what elements it is made of, pretty neat overall

i am very much a beginner myself so like the advice i can give is limited but if you have any more concrete questions ill do my best to adress them

**Reply by hlysine** [reply]
Good suggestions. Ctrl+F1 is really your best friend when you are trying to figure out how certain areas of osu work. 

Some additional points:
- Remember to read the [contributing guideline](https://github.com/ppy/osu/blob/master/CONTRIBUTING.md). There are certain rules that we follow to make the PR & review process smoother. 
- Feel free to join the [development discord server](https://discord.gg/ppy). You may get answers quicker there if you are puzzled by the codebase.
- The wiki in this repo may not help you much for now, but [osu-framework's wiki](https://github.com/ppy/osu-framework/wiki) is super helpful for knowing how fundamental things work in this game (e.g. dependency injection, what "bindables" are, how the layout system work, etc)

**Reply by Ianlucht** [reply]
As you said Yes I am a computer science major, and am pretty confident in C# through my classes in java, and a bit of self teaching for the differences in syntax. And I was thinking about going through this and looking at some of the issues as well as potential QOL stuff. I just can only do so much with that since the one area that I have not learned how to work with yet is UI, which is a bit of a bummer.

Thank you for this answer though!!
